lib/context_request_middleware/cookie.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module ContextRequestMiddleware
+  # Cookie module to consist the compatibility with
+  # rack version 1.x & 2.0
+  module Cookie
+    HTTP_COOKIE = 'HTTP_COOKIE' if Rack.release < '2.0.0'
+
+    # :nocov:
+    def session
+      if Rack.release < '2.0.0'
+        parse_cookies(@request.env)['_session_id'] ||
```

Comment:
It feels wrong to require the 'Cookie' to support an instance variable '@request'. Either make it dependent on the method 'request' or refactor so that this implicit dependency is removed.
I know why you created the module in the first place (used in two classes) but I would try to rethink this idea.